passes/cmds/wrapcell.cc

```diff
@@ -18,12 +18,37 @@
  */
 #include "kernel/yosys.h"
 #include "kernel/celltypes.h"
+#include "kernel/sigtools.h"
 #include "backends/rtlil/rtlil_backend.h"
 
 USING_YOSYS_NAMESPACE
 PRIVATE_NAMESPACE_BEGIN
 
-std::optional<std::string> format(std::string fmt, const dict<IdString, Const> &parameters)
+bool has_fmt_field(std::string fmt, std::string field_name)
+{
+	auto it = fmt.begin();
+	while (it != fmt.end()) {
+		if (*it == '{') {
+			it++;
+			auto beg = it;
+			while (it != fmt.end() && *it != '}') it++;
+			if (it == fmt.end())
+				return false;
+
+			if (std::string(beg, it) == field_name)
+				return true;
+		}
+		it++;
+	}
+	return false;
+}
+
+struct ContextData {
+	std::string unused_outputs;
+};
+
+std::optional<std::string> format(std::string fmt, const dict<IdString, Const> &parameters,
```

Comment:
Maybe mark `const`

Reply:
You mean `ContextData`? Sure, let's do that

Reply:
Sorry, wrong line. I meant mark `Chunk::format` `const`